mava/core_jax.py

```diff
@@ -0,0 +1,72 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Core Mava interfaces for Jax systems."""
+
+import abc
+from typing import Any, List
+
+
+class BaseSystem(abc.ABC):
+    """Abstract system object."""
+
+    @abc.abstractmethod
+    def design(self) -> List[Any]:
```

Comment:
I get that we add the required components and then configure them but why then do we need a design method to select certain ones only?

Reply:
Hey @AsadJeewa, the design will list the default components of the system. For example, for MADQN it will contain the components for a "base" MADQN. The `.add()` and `.update()` is then for changing the existing components or adding new ones to the base system.